### PR TITLE
Add guacscanner

### PR DESCRIPTION
**Pull request by jsf9k**
## 🗣 Description ##

This pull request modifies this Ansible role to allow for the addition of [cisagov/guacscanner-docker](https://github.com/cisagov/guacscanner-docker) to [cisagov/guacamole-composition](https://github.com/cisagov/guacamole-composition).

## 💭 Motivation and context ##

The functionality of [cisagov/guacscanner-docker](https://github.com/cisagov/guacscanner-docker) is necessary for the RPT UAT taking place in the COOL.

See also:
- cisagov/guacscanner/pull/1
- cisagov/guacscanner-docker#2
- cisagov/guacamole-composition#30
- cisagov/guacamole-packer#53
- cisagov/cool-assessment-terraform#149

## 🧪 Testing ##

I have applied these changes to env0 in our staging COOL environment and verified that they function as expected.

## ✅ Checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - _eschew scope creep!_
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated to reflect the changes in this PR.
- [x] Tests have been added and/or modified to cover the changes in this PR.
- [x] All new and existing tests pass.
- [x] Revert back to the main branch of [cisagov/guacamole-composition](https://github.com/cisagov/guacamole-composition) once cisagov/guacamole-composition#30 is merged.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -12,14 +12,15 @@ None.
...
-| optional_variable | Describe its purpose. | `default_value` | No |
-| required_variable | Describe its purpose. | n/a | Yes |
--->
+| postgres_username | The username to use when connecting to the PostgreSQL database that backends Guacamole. | n/a | Yes |
+| postgres_password | The password to use when connecting to the PostgreSQL database that backends Guacamole. | n/a | Yes |
+| private_ssh_key | The private ssh key to use for SFTP file transfer in Guacamole. | n/a | Yes |
+| rdp_username | The username for Guacamole to use when connecting to an instance via RDP. | n/a | Yes |
+| rdp_password | The password for Guacamole to use when connecting to an instance via RDP. | n/a | Yes |
+| rdp_username | The username for Guacamole to use when connecting to an instance via VNC. | n/a | Yes |
+| rdp_password | The password for Guacamole to use when connecting to an instance via VNC. | n/a | Yes |
```
Comment (dav3r) [inline README.md]: I assume this is what you meant here:
```suggestion
| vnc_username | The username for Guacamole to use when connecting to an instance via VNC. | n/a | Yes |
| vnc_password | The password for Guacamole to use when connecting to an instance via VNC. | n/a | Yes |
```

**Review by dav3r — APPROVED** [review]
👍 👍

File: molecule/debian/molecule.yml [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    privileged: yes
+    volumes:
+      - /sys/fs/cgroup:/sys/fs/cgroup:ro
+    command: /lib/systemd/systemd
+    pre_build_image: yes
+provisioner:
+  name: ansible
+  inventory:
+    host_vars:
+      debian9_systemd:
```
Comment (mcdonnnj) [inline molecule/debian/molecule.yml]: If we're not testing against Debian 9 (Stretch) is there any reason to keep all of this?

File: molecule/kali/molecule.yml [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    privileged: yes
+    volumes:
+      - /sys/fs/cgroup:/sys/fs/cgroup:ro
+    command: /lib/systemd/systemd
+    pre_build_image: yes
+provisioner:
+  name: ansible
+  inventory:
+    host_vars:
+      debian9_systemd:
```
Comment (mcdonnnj) [inline molecule/kali/molecule.yml]: Same question about Debian 9 (Stretch).

File: molecule/ubuntu/molecule.yml [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+    privileged: yes
+    volumes:
+      - /sys/fs/cgroup:/sys/fs/cgroup:ro
+    command: /lib/systemd/systemd
+    pre_build_image: yes
+provisioner:
+  name: ansible
+  inventory:
+    host_vars:
+      debian9_systemd:
```
Comment (mcdonnnj) [inline molecule/ubuntu/molecule.yml]: Third time's the charm?

**Review by mcdonnnj — APPROVED** [review]
Excellent work. I had a question about Debian 9-fu since it is not a supported platform for this role. I also went a bit too far down the rabbit hole and found that [the Dockerfile for guacamole/guacamole](https://github.com/apache/guacamole-client/blob/62b26fb594aa160478ce44c6d2bb8e6a1cef4174/guacamole-docker/bin/build-guacamole.sh#L111) is installing a PostgreSQL JDBC driver that's [over six years old](https://github.com/pgjdbc/pgjdbc/releases/tag/REL9_4_1201). We should probably make an issue in the [Apache JIRA](https://issues.apache.org/jira/projects/GUACAMOLE) to either fix it ourselves with a PR or at least get some attention on it I guess. The closest issue I could find was someone else having a problem in [GUACAMOLE-1433](https://issues.apache.org/jira/browse/GUACAMOLE-1433) and another user mentions the JDBC driver needing an update.

File: molecule/debian/molecule.yml (hunk 2 above)
Comment (jsf9k) [reply molecule/debian/molecule.yml]: The intent is to remove the Debian 9 jazz wholesale once CyHy moves on from the dark ages.  I can remove it from this particular repo.

Comment (jsf9k) [reply molecule/debian/molecule.yml]: Please see commit 57517ab.

File: molecule/kali/molecule.yml (hunk 3 above)
Comment (jsf9k) [reply molecule/kali/molecule.yml]: Please see commit 57517ab.

File: molecule/ubuntu/molecule.yml (hunk 4 above)
Comment (jsf9k) [reply molecule/ubuntu/molecule.yml]: Please see commit 57517ab.

**Comment by jsf9k** [comment]
> I also went a bit too far down the rabbit hole and found that [the Dockerfile for guacamole/guacamole](https://github.com/apache/guacamole-client/blob/62b26fb594aa160478ce44c6d2bb8e6a1cef4174/guacamole-docker/bin/build-guacamole.sh#L111) is installing a PostgreSQL JDBC driver that's [over six years old](https://github.com/pgjdbc/pgjdbc/releases/tag/REL9_4_1201). We should probably make an issue in the [Apache JIRA](https://issues.apache.org/jira/projects/GUACAMOLE) to either fix it ourselves with a PR or at least get some attention on it I guess. The closest issue I could find was someone else having a problem in [GUACAMOLE-1433](https://issues.apache.org/jira/browse/GUACAMOLE-1433) and another user mentions the JDBC driver needing an update.

Good idea.  The old driver has already bitten us.  See cisagov/guacamole-composition@8197d0e804a145a8dac67078b1e0a14c68e8e99d for more details.

**Pull request merged**